core/sql/optimizer/OptPhysRelExpr.cpp

```diff
@@ -11814,10 +11814,23 @@ void SortGroupBy::addArrangementAndOrderRequirements(
   {
     // Shouldn't/Can't add a sort order type requirement
     // if we are in DP2
-    if (rg.getStartRequirements()->executeInDP2())
-      rg.addArrangement(groupExpr(),NO_SOT);
-    else
-      rg.addArrangement(groupExpr(),ESP_SOT);
+
+   if( NOT extraOrderExpr().isEmpty())
+   {
+     ValueIdList groupExprCpy(groupExpr());
+     for (ValueId vid = extraOrderExpr().init(); extraOrderExpr().next(vid);
```

Comment:
It seems it works with partial groupby, 
As far as I understand, the shortcut groupby can only be fired if the aggreation function is min/max/anytrue, so it should not be fired as well.